### DISCUSSION TITLE
How is Frigate recording stream?

### DISCUSSION THREAD

**Discussion by aleksmastermind2** [blakeblackshear/frigate#14423] [Ask A Question]
Hello guys,
How is frigate recording streams from cameras and storing them?
Is it ffmpeg process for each camera, is process monitored, how safe it is in meaning that it will record?

**Comment by NickM-27** [comment]
Frigate uses ffmpeg to record cameras, the streams are copied directly from the camera and stored in 10 second mp4 segments

**Comment by hawkeye217** [comment]
Yes, Frigate monitors ffmpeg constantly. If it fails, it will attempt to reconnect.

**Comment by aleksmastermind2** [comment]
Did anyone test recording 200+ cameras?

**Reply by NickM-27** [reply]
there are many users running 30+ cameras, we are aware of some users with ~100+ cameras

**Reply by aleksmastermind2** [reply]
Is it stable?
Can it be used for enterprise environment?
Also, can i sell this as modified version?

**Reply by NickM-27** [reply]
not sure what you mean by stable, many people use frigate. We don't have analytics to say how these systems are working we just know from talking to users. 

you should read the license https://github.com/blakeblackshear/frigate#MIT-1-ov-file

**Reply by aleksmastermind2** [reply]
Thanks very much. I really appreciate the help.
